dotnet/src/Agents/Core/Chat/ChatExecutionSettings.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Agents.Chat;
+
+/// <summary>
+/// Delegate definition for <see cref="ChatExecutionSettings.TerminationStrategy"/>.
+/// </summary>
+/// <param name="agent">The agent actively interacting with the nexus.</param>
+/// <param name="history">The chat history.</param>
+/// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+/// <returns>True to terminate chat loop.</returns>
+public delegate Task<bool> TerminationCriteriaCallback(Agent agent, IReadOnlyList<ChatMessageContent> history, CancellationToken cancellationToken);
+
+/// <summary>
+/// Delegate definition for <see cref="ChatExecutionSettings.SelectionStrategy"/>.
+/// </summary>
+/// <param name="agents">The agents participating in chat.</param>
+/// <param name="history">The chat history.</param>
+/// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+/// <returns>The agent who shall take the next turn.</returns>
+public delegate Task<Agent?> SelectionCriteriaCallback(IReadOnlyList<Agent> agents, IReadOnlyList<ChatMessageContent> history, CancellationToken cancellationToken);
+
+/// <summary>
+/// Settings that affect behavior of <see cref="AgentChat"/>.
+/// </summary>
+/// <remarks>
+/// Default behavior result in no agent selection.
+/// </remarks>
+public class ChatExecutionSettings
+{
+    /// <summary>
+    /// Restrict number of turns to one, by default.
+    /// </summary>
+    public const int DefaultMaximumIterations = 1;
```

Comment:
Most conservative value for someone that hasn't explicitly opted into.  I think this may have also came up in a different context.  Anyway, the tracking issue is:

https://github.com/microsoft/semantic-kernel/issues/5844

Happy to set to whatever we decide is best.

Reply:
FYI - This has been relocated to the `TerminationStrategy`